### PR TITLE
Skip secureboot and nvdimm testing for guest OS not supporting them

### PR DESCRIPTION
**Pull request by keirazhang**
Skip secureboot_enable_disable for Pardus, UOS, ProLinux.
Skip nvdimm_cold_add_remove for UOS, Amazon Linux and Ubuntu Cloud Image.

### CODE REVIEW

**Comment by keirazhang** [comment]
Test results:
```
  - 'Guest os distribution: Ubuntu 20.04 CloudImage x86_64'
  - 'Hardware version: vmx-21'
  - 'Config guest id: ubuntu64Guest'
  - 'Guestinfo guest id: ubuntu64Guest'
  - 'Guestinfo guest full name: Ubuntu Linux (64-bit)'
  - 'Guestinfo guest family: linuxGuest'
  - 'Guestinfo detailed data: architecture=''X86'' bitness=''64'' distroName=''Ubuntu''
    distroVersion=''20.04'' familyName=''Linux'' kernelVersion=''5.4.0-139-generic''
    prettyName=''Ubuntu 20.04.5 LTS'''
  - 'Results: Total (4), Passed (2), Skipped (2)'
  - ' * deploy_vm_ova...................................................Passed'
  - ' * secureboot_enable_disable...............................Not Applicable'
  - ' * nvdimm_cold_add_remove...................................Not Supported'
  - ' * check_os_fullname...............................................Passed'


  - 'Guest os distribution: Amazon 2 x86_64'
  - 'Hardware version: vmx-21'
  - 'Config guest id: other3xLinux64Guest'
  - 'Guestinfo guest id: amazonlinux2_64Guest'
  - 'Guestinfo guest full name: Amazon Linux 2 (64-bit)'
  - 'Guestinfo guest family: linuxGuest'
  - 'Guestinfo detailed data: architecture=''X86'' bitness=''64'' distroName=''Amazon
    Linux'' distroVersion=''2'' familyName=''Linux'' kernelVersion=''4.14.301-224.520.amzn2.x86_64''
    prettyName=''Amazon Linux 2'''
  - 'Results: Total (4), Passed (2), Skipped (2)'
  - ' * deploy_vm_ova...................................................Passed'
  - ' * secureboot_enable_disable...............................Not Applicable'
  - ' * nvdimm_cold_add_remove...................................Not Supported'
  - ' * check_os_fullname...............................................Passed'


  - 'Guest os distribution: Ubuntu 23.04 CloudImage x86_64'
  - 'Hardware version: vmx-21'
  - 'Config guest id: ubuntu64Guest'
  - 'Guestinfo guest id: ubuntu64Guest'
  - 'Guestinfo guest full name: Ubuntu Linux (64-bit)'
  - 'Guestinfo guest family: linuxGuest'
  - 'Guestinfo detailed data: architecture=''X86'' bitness=''64'' distroName=''Ubuntu
    23.04'' distroVersion=''23.04'' familyName=''Linux'' kernelVersion=''6.2.0-20-generic''
    prettyName=''Ubuntu 23.04'''
  - 'Results: Total (4), Passed (2), Skipped (2)'
  - ' * deploy_vm_ova...................................................Passed'
  - ' * secureboot_enable_disable...............................Not Applicable'
  - ' * nvdimm_cold_add_remove...................................Not Supported'
  - ' * check_os_fullname...............................................Passed'


  - 'Guest os distribution: Ubuntu 22.04 CloudImage x86_64'
  - 'Hardware version: vmx-21'
  - 'Config guest id: ubuntu64Guest'
  - 'Guestinfo guest id: ubuntu64Guest'
  - 'Guestinfo guest full name: Ubuntu Linux (64-bit)'
  - 'Guestinfo guest family: linuxGuest'
  - 'Guestinfo detailed data: architecture=''X86'' bitness=''64'' distroName=''Ubuntu
    22.04 LTS'' distroVersion=''22.04'' familyName=''Linux'' kernelVersion=''5.15.0-25-generic''
    prettyName=''Ubuntu 22.04 LTS'''
  - 'Results: Total (4), Passed (2), Skipped (2)'
  - ' * deploy_vm_ova...................................................Passed'
  - ' * secureboot_enable_disable...............................Not Applicable'
  - ' * nvdimm_cold_add_remove...................................Not Supported'
  - ' * check_os_fullname...............................................Passed'


  - 'Guest os distribution: Ubuntu 23.04 Desktop x86_64'
  - 'Hardware version: vmx-20'
  - 'Config guest id: ubuntu64Guest'
  - 'Guestinfo guest id: ubuntu64Guest'
  - 'Guestinfo guest full name: Ubuntu Linux (64-bit)'
  - 'Guestinfo guest family: linuxGuest'
  - 'Guestinfo detailed data: architecture=''X86'' bitness=''64'' distroName=''Ubuntu
    23.04'' distroVersion=''23.04'' familyName=''Linux'' kernelVersion=''6.2.0-20-generic''
    prettyName=''Ubuntu 23.04'''
  - 'Results: Total (4), Passed (4)'
  - ' * deploy_vm_efi_paravirtual_vmxnet3...............................Passed'
  - ' * secureboot_enable_disable.......................................Passed'
  - ' * nvdimm_cold_add_remove..........................................Passed'
  - ' * check_os_fullname...............................................Passed'


  - 'Guest os distribution: ProLinux 8.5 x86_64'
  - 'Hardware version: vmx-20'
  - 'Config guest id: other4xLinux64Guest'
  - 'Guestinfo guest id: other4xLinux64Guest'
  - 'Guestinfo guest full name: Other 4.x Linux (64-bit)'
  - 'Guestinfo guest family: linuxGuest'
  - 'Guestinfo detailed data: bitness=''64'' distroName=''ProLinux'' distroVersion=''8.5''
    familyName=''Linux'' kernelVersion=''4.18.0-348.el8.x86_64'' prettyName=''ProLinux
    8.5'''
  - 'Results: Total (4), Passed (3), Skipped (1)'
  - ' * deploy_vm_efi_paravirtual_vmxnet3...............................Passed'
  - ' * secureboot_enable_disable................................Not Supported'
  - ' * nvdimm_cold_add_remove..........................................Passed'
  - ' * check_os_fullname...............................................Passed'


  - 'Guest os distribution: Ubuntu 20.04 Server x86_64'
  - 'Hardware version: vmx-20'
  - 'Config guest id: ubuntu64Guest'
  - 'Guestinfo guest id: ubuntu64Guest'
  - 'Guestinfo guest full name: Ubuntu Linux (64-bit)'
  - 'Guestinfo guest family: linuxGuest'
  - 'Guestinfo detailed data: architecture=''X86'' bitness=''64'' distroName=''Ubuntu''
    distroVersion=''20.04'' familyName=''Linux'' kernelVersion=''5.4.0-144-generic''
    prettyName=''Ubuntu 20.04.6 LTS'''
  - 'Results: Total (4), Passed (4)'
  - ' * deploy_vm_efi_paravirtual_vmxnet3...............................Passed'
  - ' * secureboot_enable_disable.......................................Passed'
  - ' * nvdimm_cold_add_remove..........................................Passed'
  - ' * check_os_fullname...............................................Passed'

  - 'Guest os distribution: ProLinux 7.9 x86_64'
  - 'Hardware version: vmx-20'
  - 'Config guest id: other3xLinux64Guest'
  - 'Guestinfo guest id: other3xLinux64Guest'
  - 'Guestinfo guest full name: Other 3.x Linux (64-bit)'
  - 'Guestinfo guest family: linuxGuest'
  - 'Guestinfo detailed data: bitness=''64'' distroName=''ProLinux Server'' distroVersion=''7.9''
    familyName=''Linux'' kernelVersion=''3.10.0-1160.el7.x86_64'' prettyName=''ProLinux
    7.9'''
  - 'Results: Total (4), Passed (3), Skipped (1)'
  - ' * deploy_vm_efi_paravirtual_vmxnet3...............................Passed'
  - ' * secureboot_enable_disable................................Not Supported'
  - ' * nvdimm_cold_add_remove..........................................Passed'
  - ' * check_os_fullname...............................................Passed'

  - 'Guest os distribution: Ubuntu 22.04 Desktop x86_64'
  - 'Hardware version: vmx-20'
  - 'Config guest id: ubuntu64Guest'
  - 'Guestinfo guest id: ubuntu64Guest'
  - 'Guestinfo guest full name: Ubuntu Linux (64-bit)'
  - 'Guestinfo guest family: linuxGuest'
  - 'Guestinfo detailed data: architecture=''X86'' bitness=''64'' distroName=''Ubuntu
    22.04.2 LTS'' distroVersion=''22.04'' familyName=''Linux'' kernelVersion=''5.19.0-43-generic''
    prettyName=''Ubuntu 22.04.2 LTS'''
  - 'Results: Total (4), Passed (4)'
  - ' * deploy_vm_efi_paravirtual_vmxnet3...............................Passed'
  - ' * secureboot_enable_disable.......................................Passed'
  - ' * nvdimm_cold_add_remove..........................................Passed'
  - ' * check_os_fullname...............................................Passed'


  - 'Guest os distribution: Ubuntu 23.04 Server x86_64'
  - 'Hardware version: vmx-20'
  - 'Config guest id: ubuntu64Guest'
  - 'Guestinfo guest id: ubuntu64Guest'
  - 'Guestinfo guest full name: Ubuntu Linux (64-bit)'
  - 'Guestinfo guest family: linuxGuest'
  - 'Guestinfo detailed data: architecture=''X86'' bitness=''64'' distroName=''Ubuntu
    23.04'' distroVersion=''23.04'' familyName=''Linux'' kernelVersion=''6.2.0-20-generic''
    prettyName=''Ubuntu 23.04'''
  - 'Testset folder: Root\ALPS\Keira\Ansible\GOSV_Async_Pipeline\1334_20230605-033844'
  - 'Testset name: Ansible_Regression_Ubuntu_23.04_Server_ISO'
  - 'Testset id: 1238095'
  - 'Results: Total (4), Passed (4)'
  - ' * deploy_vm_efi_paravirtual_vmxnet3...............................Passed'
  - ' * secureboot_enable_disable.......................................Passed'
  - ' * nvdimm_cold_add_remove..........................................Passed'
  - ' * check_os_fullname...............................................Passed'


  - 'Guest os distribution: UnionTech 20 x86_64'
  - 'Hardware version: vmx-20'
  - 'Config guest id: other4xLinux64Guest'
  - 'Guestinfo guest id: other5xLinux64Guest'
  - 'Guestinfo guest full name: Other 5.x Linux (64-bit)'
  - 'Guestinfo guest family: linuxGuest'
  - 'Guestinfo detailed data: architecture=''X86'' bitness=''64'' distroName=''UnionTech
    OS Server 20'' distroVersion=''20'' familyName=''Linux'' kernelVersion=''5.10.0-18.uelc20.x86_64''
    prettyName=''UnionTech OS Server 20'''
  - 'Results: Total (4), Passed (2), Skipped (2)'
  - ' * deploy_vm_efi_paravirtual_vmxnet3...............................Passed'
  - ' * secureboot_enable_disable................................Not Supported'
  - ' * nvdimm_cold_add_remove...................................Not Supported'
  - ' * check_os_fullname...............................................Passed'


  - 'Guest os distribution: Ubuntu 22.04 Server x86_64'
  - 'Hardware version: vmx-20'
  - 'Config guest id: ubuntu64Guest'
  - 'Guestinfo guest id: ubuntu64Guest'
  - 'Guestinfo guest full name: Ubuntu Linux (64-bit)'
  - 'Guestinfo guest family: linuxGuest'
  - 'Guestinfo detailed data: architecture=''X86'' bitness=''64'' distroName=''Ubuntu
    22.04.2 LTS'' distroVersion=''22.04'' familyName=''Linux'' kernelVersion=''5.15.0-73-generic''
    prettyName=''Ubuntu 22.04.2 LTS'''
  - 'Results: Total (4), Passed (4)'
  - ' * deploy_vm_efi_paravirtual_vmxnet3...............................Passed'
  - ' * secureboot_enable_disable.......................................Passed'
  - ' * nvdimm_cold_add_remove..........................................Passed'
  - ' * check_os_fullname...............................................Passed'
```

File: linux/nvdimm_cold_add_remove/nvdimm_cold_add_remove.yml [hunk 1]
```diff
@@ -17,11 +17,13 @@
 
         - include_tasks: ../../common/skip_test_case.yml
           vars:
-            skip_msg: "{{ guest_os_ansible_distribution }} doesn't support NVDIMM"
+            skip_msg: "{{ vm_guest_os_distribution }} doesn't support NVDIMM"
             skip_reason: "Not Supported"
           when: >
             ('Flatcar' in guest_os_ansible_distribution or
-             guest_os_ansible_distribution == 'UnionTech')
+             guest_os_ansible_distribution in ['UnionTech', 'Uos', 'Amazon'] or
```
Comment (123lzxm) [inline linux/nvdimm_cold_add_remove/nvdimm_cold_add_remove.yml:24]: Flatcar can be also put in the list.

Comment (keirazhang) [reply linux/nvdimm_cold_add_remove/nvdimm_cold_add_remove.yml:24]: Flatcar's distribution might be changed from version to version. But it always has 'Flatcar' in it.

**Review by 123lzxm — APPROVED** [review]
LGTM except one minor comment. Thanks.

File: linux/nvdimm_cold_add_remove/nvdimm_cold_add_remove.yml [hunk 2]
```diff
@@ -42,12 +54,15 @@
           vars:
             guest_id: "{{ vm_guest_id }}"
             esxi_hardware_version: "{{ vm_hardware_version_num }}"
-        - include_tasks: ../../common/skip_test_case.yml
+
+        - name: "Skip test case for not supported guest id"
```
[the remark this answers is not in the document]
Comment (keirazhang) [reply linux/nvdimm_cold_add_remove/nvdimm_cold_add_remove.yml]: Updated. Thanks!

**Review by 123lzxm — APPROVED** [review]
LGTM, thanks.

**Pull request merged**